CRM/Contact/BAO/Query.php

```diff
@@ -6159,62 +6166,64 @@ protected function prepareOrderBy($sort, $sortByChar, $sortOrder, $additionalFro
         }
       }
       elseif ($sortByChar) {
-        $order = " ORDER BY UPPER(LEFT(contact_a.sort_name, 1)) asc";
+        $orderByArray = array("UPPER(LEFT(contact_a.sort_name, 1)) asc");
       }
       else {
         $order = " ORDER BY contact_a.sort_name asc, contact_a.id";
       }
     }
+    if (!$order && empty($orderByArray)) {
```

Comment:
in all of this I'm trying to move towards not compiling the string until the end - although this is only part transitionedl